src/backends/beamcoder.ts

```diff
@@ -210,11 +217,8 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
      */
     get duration(): number {
         const time_base = this.#demuxer.streams[this.#streamIndex].time_base;
-        const durations = this.#demuxer.streams.map(
-            stream => stream.duration * time_base[0] / time_base[1]
-        );
-
-        return Math.max(...durations);
+        const duration = this.#demuxer.streams[this.#streamIndex].duration * time_base[0] / time_base[1];
```

Comment:
We could use `ptsToTime` to simplify this function now `duration = this.ptsToTime(this.#demuxer.streams[this.#streamIndex].duration)`